packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -60,12 +66,48 @@ export function tryGetTreeNodeSchema(value: unknown): undefined | TreeNodeSchema
  * The kernel has the same lifetime as the node and spans both its unhydrated and hydrated states.
  * When hydration occurs, the kernel is notified via the {@link TreeNodeKernel.hydrate | hydrate} method.
  */
-export class TreeNodeKernel implements Listenable<TreeChangeEvents> {
+export class TreeNodeKernel implements Listenable<KernelEvents> {
+	private disposed = false;
+
+	/**
+	 * Generation number which is incremented any time we have an edit on the node.
+	 * Used during iteration to make sure there has been no edits that were concurrently made.
+	 */
+	public generationNumber: number = 0;
```

Comment:
Should mutations to this be more controlled? E.g., make the setter private and expose an `increment` method - otherwise consumers could potentially make invalid changes to this.

Reply:
Personally for very internal implementation detail code like this (not even exported out of simple-tree) I don't think its worth the extra boiler plate, runtime overhead and bundle size to make that change. If someday we want a bit more encapsulation here we can make a TreeNodeKernel interface so we can have that improved safety without the runtime or bundle size overhead.

A decent portion of fluid runtime bundle size is wrappers around class properties to prevent external writing (they are a few percent of one of our largest files, see packages/runtime/container-runtime/src/containerRuntime.ts): I don't think that pattern is worth the cost here.